Fix Shift-JIS throwing Unsupported operand Error

askdkc opened the pull request:
This will fix ` private static function isOnlyDoubleByteKanji(string $content) : bool` throwing "Unsupported operand types: string & int" when using SHIFT-JIS encoding.

<img width="800" alt="image" src="https://user-images.githubusercontent.com/7894265/211293518-fa865c4d-c951-4c5b-af07-5085ad6b9c2d.png">

DASPRiD commented:
Concatenation is definitely incorrect here. Looking at the original Java code it did the bitwise operation, but since it's always a single byte, the correct solution would probably be `$byte = ord($bytes[$i]);`. This would require some testing though, as this is currently not covered by unit tests (any fix should add unit tests for that).

askdkc commented:
@DASPRiD 
OK, thanks for your comment.

I fixed the code and added test for `isOnlyDoubleByteKanji()`.

Everybody is using UTF-8 these days, and if you are scanning QRcode with your smart phone, it works fine. But cheap QRcode Readers selling in Japan only support SHIFT-JIS encoded QRcode and Windows PC system still uses SHIFT-JIS encoding for Japanese.

BaconQrCode Ver 1 is working fine, but ver 2 having this issue prevents upgrading from ver 1. This problem makes some system unable to upgrade not only BaconQrCode but php itself, that's not good.

Hopefully this fix gets merged and makes Japanese people happy🇯🇵🗻

askdkc commented:
@DASPRiD 

I saw some Lint errors in CI, so I fixed them.
Hopefully this PR gets merged and soon to be able to use SHIFT-JIS encoded QRcode again😃

askdkc commented:
Hmm, after I apply this fix to the package, `appendKanjiBytes(string $content, BitArray $bits)` starts throwing an error.

When I generate QRcode with Kanji and ASCII mix characters, it works fine. For example "あいうえお123" can correctly generate QRcode.

But if I try to generate QRcode that contains only double byte Kanji like "日本" it throws this error.

Is there something wrong with `appendKanjiBytes(string $content, BitArray $bits)` function🤔

<img width="1275" alt="image" src="https://user-images.githubusercontent.com/7894265/213619290-b544f094-b19e-4cf7-a77c-068847a6364b.png">

askdkc commented:
This PR was kinda recreated to [#124](https://github.com/Bacon/BaconQrCode/pull/124)
